posts/2021-01-06-ocp-log-forwarding.adoc

```diff
@@ -0,0 +1,425 @@
++
+```
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogForwarder
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  disableDefaultForwarding: true
+  outputs:
+   - name: fluentd-server-secure
+     type: fluentdForward
+     url: 'tls://fluntdserver.security.example.com:24224' # Substitute with FQDN or IP address
+     secret:
+        name: secure-forward
+   - name: fluentd-server-insecure
+     type: fluentdForward
+     url: 'tcp://fluntdserver.home.example.com:24224' # Substitute with FQDN or IP address
+  pipelines:
+   - name: container-logs
+     inputRefs:
+     - application
+     outputRefs:
+     - default
+     - fluentd-server-secure
+   - name: infra-logs
+     inputRefs:
+     - infrastructure
+     outputRefs:
+     - default
+   - name: audit-logs
+     inputRefs:
+     - audit
+     outputRefs:
+     - default
+
+```
++
+For more information, see link:https://docs.openshift.com/container-platform/4.6/logging/cluster-logging-external.html#cluster-logging-collector-log-forward-fluentd_cluster-logging-external[Forwarding logs using the Fluentd forward protocol]. Make sure to consult the documentation for your version of OpenShift, as the pipeline configuration for version 4.6 is slightly different from the preceding versions.
+
+
+[start=2]
+. Create the Log Forwarding instance inside your OpenShift cluster:
++
+[source]
+----
+[root@ocp ~]# oc create -f log-forward-instance.yaml
+----
++
+
+. **OpenShift 4.3 to 4.5 only:** Annotate the ClusterLogging instance to enable the Log Forwarding API.
++
+[source]
+----
+[root@ocp ~]# oc annotate clusterlogging -n openshift-logging instance clusterlogging.openshift.io/logforwardingtechpreview=enabled
+----
++
+
+[start=3]
+. To check whether the logs are forwarded to the specified outputs, run the following command:
++
+[source]
+----
+[root@ocp ~]# oc -n openshift-logging get cm fluentd -o json | jq -r '.data."fluent.conf"' > fluentd-with-logfowarding.conf
+----
++
+This command displays the ConfigMap configuration for Fluentd inside OpenShift. Check whether the outputs are defined inside the configuration file.
+
+* In the following configuration file example, you can see the `elasticsearch` and `fluent-forward` outputs that are referenced by `@label`:
++
+```
+...
+<label @CONTAINER_LOGS>
+  <match **>
+    @type copy
+
+    <store>
+      @type relabel
+      @label @DEFAULT
+    </store>
+    <store>
+      @type relabel
+      @label @FLUENTD_FORWARD
+    </store>
+  </match>
+</label>
+...
+```
++
+**OpenShift 4.3 to 4.5 only:**  The value `@DEFAULT` would be `@ELASTICSEARCH`
```

Comment:
suggested edit:

**OpenShift 4.3 to 4.5 only:**  The `@label @DEFAULT` value is replaced by `@label @ELASTICSEARCH`.